frontend/src/libs/components/protected-route/protected-route.tsx

```diff
@@ -0,0 +1,17 @@
+import { AppRoute } from '~/libs/enums/enums.js';
+import { Navigate, useAppSelector } from '~/libs/hooks/hooks.js';
+
+type Properties = {
+  children: React.ReactNode;
+};
+const ProtectedRoute: React.FC<Properties> = ({ children }: Properties) => {
+  const { user } = useAppSelector(({ auth }) => ({ user: auth.user }));
+  const hasAuthenticatedUser = Boolean(user);
+  if (!hasAuthenticatedUser) {
+    return <Navigate to={AppRoute.SIGN_IN} replace />;
+  }
```

Comment:
There is build-in type declared for `children` prop
```suggestion
const ProtectedRoute: React.FC<React.PropsWithChildren> = ({ children }: React.PropsWithChildren) => {
  const { user } = useAppSelector(({ auth }) => ({ user: auth.user }));
  const hasAuthenticatedUser = Boolean(user);
  
  if (!hasAuthenticatedUser) {
    return <Navigate to={AppRoute.SIGN_IN} replace />;
  }
```